core/shared/src/main/scala-3/monocle/internal/focus/features/SelectParserBase.scala

```diff
@@ -26,16 +26,24 @@ private[focus] trait SelectParserBase extends ParserBase {
     case None      => FocusError.NotAConcreteClass(tpe.show).asResult
   }
 
-  def getFieldType(fromType: TypeRepr, fieldName: String): FocusResult[TypeRepr] = {
-    // We need to do this to support tuples, because even though they conform as case classes in other respects,
-    // for some reason their field names (_1, _2, etc) have a space at the end, ie `_1 `.
-    def getTrimmedFieldSymbol(fromTypeSymbol: Symbol): Symbol =
-      fromTypeSymbol.memberFields.find(_.name.trim == fieldName).getOrElse(Symbol.noSymbol)
+  private val tupleFieldPattern = "^_[0-9]+$".r
+
+  def getFieldType(fromType: TypeRepr, fieldName: String, pos: Position): FocusResult[TypeRepr] = {
+    def getFieldSymbol(fromTypeSymbol: Symbol): Symbol = {
+      // We need to do this to support tuples, because even though they conform as case classes in other respects,
+      // for some reason their field names (_1, _2, etc) have a space at the end, ie `_1 `.
+      val f: String => String =
```

Comment:
This fixes a silent error if we have something like

```scala case class Foo(x: Int, `x `: Int)``` since identifiers with leading/trailing spaces are not equivalent.